middleware/locals.js

```diff
@@ -57,6 +58,44 @@ module.exports = function(req, res, next) {
         };
     });
 
+    /**
+     * Global navigation model
+     */
+    res.locals.globalNavigation = {
```

Comment:
Given how one-off the global navigation is I've gone with the approach of explicitly modelling the structure here rather than computing it from the routes file. There's probably a way to simplify some of the boilerplate here, but I think the concept makes sense.